From e1f3c162f48a8ba7fbb06b930c79df128c80e096 Mon Sep 17 00:00:00 2001
From: cubewhy <61075476+cubewhy@users.noreply.github.com>
Date: Tue, 6 Jan 2026 18:02:32 +0800
Subject: [PATCH] fix: Distrobox is not location independent on NixOS

---
 distrobox-create | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/distrobox-create b/distrobox-create
index 72b87d8748..e63e4f99f3 100755
--- a/distrobox-create
+++ b/distrobox-create
@@ -96,6 +96,15 @@ distrobox_entrypoint_path="$(cd "$(dirname "${0}")" && pwd)/distrobox-init"
 distrobox_export_path="$(cd "$(dirname "${0}")" && pwd)/distrobox-export"
 distrobox_genentry_path="$(cd "$(dirname "${0}")" && pwd)/distrobox-generate-entry"
 distrobox_hostexec_path="$(cd "$(dirname "${0}")" && pwd)/distrobox-host-exec"
+
+# NixOS specific path definition
+nixos_bin_path="/run/current-system/sw/bin"
+if [ -d "${nixos_bin_path}" ]; then
+	[ ! -e "${distrobox_entrypoint_path}" ] && [ -e "${nixos_bin_path}/distrobox-init" ] && distrobox_entrypoint_path="${nixos_bin_path}/distrobox-init"
+	[ ! -e "${distrobox_export_path}" ] && [ -e "${nixos_bin_path}/distrobox-export" ] && distrobox_export_path="${nixos_bin_path}/distrobox-export"
+	[ ! -e "${distrobox_genentry_path}" ] && [ -e "${nixos_bin_path}/distrobox-generate-entry" ] && distrobox_genentry_path="${nixos_bin_path}/distrobox-generate-entry"
+	[ ! -e "${distrobox_hostexec_path}" ] && [ -e "${nixos_bin_path}/distrobox-host-exec" ] && distrobox_hostexec_path="${nixos_bin_path}/distrobox-host-exec"
+fi
 # In case some of the scripts are not in the same path as create, let's search
 # in PATH for them.
 [ ! -e "${distrobox_entrypoint_path}" ] && distrobox_entrypoint_path="$(command -v distrobox-init)"
